mongomock/aggregate.py

```diff
@@ -315,7 +315,16 @@ def _handle_project_operator(self, operator, values):
         if operator == '$arrayElemAt':
             key, index = values
             array = self._parse_basic_expression(key)
-            return array[index]
+
+            if index < len(array):
+                return array[index]
```

Comment:
What about negative values? Can you add a test for that as well please?

I guess the implementation could be easier and more pythonic with:
```py
try:
  return array[index]
except IndexError:
  raise KeyError('%s is not a valid index in array' % index)
```